### PR TITLE
TRUNK-5890: Use annotations to configure Privilege entity

### PR DESCRIPTION
**Pull request by abertnamanya**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5890

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -517,7 +517,7 @@
 			<dependency>
 				<groupId>net.sf.ehcache</groupId>
 				<artifactId>ehcache</artifactId>
-				<version>2.10.10.1.14</version>
+				<version>2.10.9.2</version>
```
Comment (sherrif10) [inline pom.xml:520]: Any reason of  downgrading ehcache version ?

Comment (abertnamanya) [reply pom.xml:520]: when I made git pull --rebase upstream master after creating the branch  dependabot  had added  2.10.10.1.14 on remote master which wasn't working on the ehcache imports (even 2.10.10.1.14 v seems not yet out . link https://mvnrepository.com/artifact/net.sf.ehcache/ehcache)

File: api/src/main/java/org/openmrs/Privilege.java [hunk 2]
```diff
@@ -9,21 +9,29 @@
...
+@AttributeOverride(name = "description", column = @Column(name = "description", length = 250))
 public class Privilege extends BaseChangeableOpenmrsMetadata {
 	
 	public static final long serialVersionUID = 312L;
 	
-	// Fields
-	
+	@Id
+	@Column(name = "privilege",nullable = false,length = 250)
 	private String privilege;
```
Comment (HerbertYiga) [inline api/src/main/java/org/openmrs/Privilege.java:33]: apart from privilege, do we have other attributes defined as  required!!

Comment (abertnamanya) [reply api/src/main/java/org/openmrs/Privilege.java:33]: Nop, its just privilege,description and uuid

**Comment by tendomart** [comment]
@abertnamanya hope you're looking into the failed tests

**Pull request closed without merging**